src/ray/object_manager/pull_manager.cc

```diff
@@ -183,7 +184,9 @@ void PullManager::UpdatePullsBasedOnAvailableMemory(size_t num_bytes_available)
     }
 
     // Activate the next worker request if we have space.
-    if (num_bytes_being_pulled_ < num_bytes_available_) {
+    // TODO(ekl) in unlimited mode, this should trigger spilling on OOM?
```

Comment:
@stephanie-wang do the objects created by activating the pull bundle end up going through CreateRequestQueue? (i.e., would we trigger spilling here automatically if needed?)

Reply:
A more general question--- is there any allocation path that doesn't go through the CreateRequestQueue? I couldn't seem to find any.

Reply:
All objects should go through the queue, so it should be okay.

Reply:
Hmm actually isn't this change a bit fishy? I think we still want to keep under the threshold even if plasma is unlimited.

Reply:
This is needed for just the ray.get() requests--- otherwise we'd hang. The queueing behavior for task arg requests is still preserved.

Reply:
Ray.get is unlimited, there is never any throttling here.

Reply:
I think @rkooo567 is right, ray.get should get throttled by the pull threshold.

Reply:
That's the main point of the PR though, to avoid deadlock on get. Throttling a get is always going to be blocking user code.

We could potentially throttle wait(). Do you know how wait requests translate into PullManager requests right now?

Note that even if not throttled, these requests still count towards memory usage and can in turn throttle task launches.